File: src/dlls/CSSPModels/srcweb/WebSamplingPlan.cs
﻿/*
 * Manually edited
 * 
 */
using System.Collections.Generic;

namespace CSSPModels
{
    public partial class WebSamplingPlan
    {
        #region Properties
        public SamplingPlanModel SamplingPlanModel { get; set; }
        #endregion Properties

        #region Constructors
        public WebSamplingPlan()
        {
            SamplingPlanModel = new SamplingPlanModel();
        }
        #endregion Constructors
    }

    public partial class SamplingPlanModel
    {
        #region Properties
        public SamplingPlan SamplingPlan { get; set; }
        public List<SamplingPlanSubsectorModel> SamplingPlanSubsectorModelList { get; set; }
        public List<SamplingPlanEmail> SamplingPlanEmailList { get; set; }
        #endregion Properties

        #region Constructors
        public SamplingPlanModel()
        {
            SamplingPlanSubsectorModelList = new List<SamplingPlanSubsectorModel>();
            SamplingPlanEmailList = new List<SamplingPlanEmail>();
        }
        #endregion Constructors
    }

    public partial class SamplingPlanSubsectorModel
    {
        #region Properties
        public SamplingPlanSubsector SamplingPlanSubsector { get; set; }
        public List<SamplingPlanSubsectorSite> SamplingPlanSubsectorSiteList { get; set; }
        #endregion Properties

        #region Constructors
        public SamplingPlanSubsectorModel()
        {
            SamplingPlanSubsector = new SamplingPlanSubsector();
            SamplingPlanSubsectorSiteList = new List<SamplingPlanSubsectorSite>();
        }
        #endregion Constructors
    }
}

File: src/webs/CSSPWebToolsAng/client-apps/src/app/models/websamplingplan.ts
import { SamplingPlan } from './generated/SamplingPlan.model';
import { SamplingPlanSubsector } from './generated/SamplingPlanSubsector.model';
import { SamplingPlanSubsectorSite } from './generated/SamplingPlanSubsectorSite.model';
import { WebBase } from './webbase';

export class WebSamplingPlan {
    TVItemParentList: WebBase[] = [];
    SamplingPlanModelList: SamplingPlanModel[] = [];
}

export class SamplingPlanModel {
    SamplingPlan: SamplingPlan = new SamplingPlan();
    SamplingPlanSubsectorModelList: SamplingPlanSubsectorModel[] = [];
}

export class SamplingPlanSubsectorModel {
    SamplingPlanSubsector: SamplingPlanSubsector = new SamplingPlanSubsector();
    SamplingPlanSubsectorSiteList: SamplingPlanSubsectorSite[] = [];
}